ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -378,6 +381,32 @@ short getVersion() {
     return toReturn != null ? Collections.unmodifiableNavigableSet(toReturn) : null;
   }
 
+  /**
+   * Generate bloom file by walking through all index entries in this segment and persist it.
+   * @throws IOException
+   */
+  private void generateBloomFileAndPersist() throws IOException {
+    List<IndexEntry> entries = new ArrayList<>();
+    getIndexEntriesSince(null, new FindEntriesCondition(Long.MAX_VALUE), entries, new AtomicLong(0), true);
+    bloomFilter = FilterFactory.getFilter(entries.size(), config.storeIndexBloomMaxFalsePositiveProbability);
+    for (IndexEntry entry : entries) {
+      bloomFilter.add(ByteBuffer.wrap(entry.getKey().toBytes()));
+    }
+    persistBloomFile();
+  }
+
+  /**
+   * Persist the bloom file.
+   * @throws IOException
+   */
+  private void persistBloomFile() throws IOException {
```

Comment:
I wonder if "persist" is the right term here. "persist" in the context of the store usually means a flush to the disk, which is not the case here. This is really only about creating the file. I know this is the term we used perviously, but does it make sense to call this "createBloomFile"?

Reply:
"persist" might be slightly ambiguous here but it makes the description in code consistent, doesn't it? @vgkholla what do you think of it?

Reply:
@pnarayanan Not sure I understand - this piece of code is actually persisting the file except that it doesn't "force" the flush (and it always had a bug of not closing the stream). We can add both to make it clearer - but it's doing the persist, so the naming is correct, I think.

Reply:
The bloom filter file for a segment is only created and written to once - previously at the time it becomes a mapped segment, and now in the recovery case. And that is what is being called as persist in this context - as in, create a "disk" representation.

This is different from the "persist" in the IndexPersistor (which is all about the flushing).

Reply:
I don't think `IndexPersistor` is all about flushing - The `IndexPersistor` creates the `IndexSegment` file too (and the bloom filter file too in the context of making it a mapped segment) i.e. it does more than just flush existing files.

Reply:
It is mostly about syncing data to disk, so at least in my mind the "persisting" is about flushing - creation happens occasionally, but the reason the persistor runs every minute or so is to flush in-memory data to disk.

I am fine with leaving the name unchanged.